addon-test-support/load.js

```diff
@@ -19,4 +22,73 @@ export default function loadEmberExam() {
   if (window.Testem) {
     patchTestemOutput(TestLoader);
   }
+
+  testLoader = new TestLoader();
 }
+
+// loadTests() is equivalent to ember-qunit's loadTest() except this does not create a new TestLoader instance
+function loadTests() {
+  if (testLoader === undefined) {
+    new Error('testLoader is not defined. It must call `loadEmberExam()` before calling `loadTest()`.')
+  }
+
+  testLoader.loadModules();
+}
+
+// setupQUnitCallBacks() registers QUnit callbacks neeeded for the load-balance option.
+function setupQUnitCallbacks(qunit) {
+  if (!location.search.includes('loadBalance')) {
+    return;
+  }
+  qunit.begin(function() {
+    return new self.Promise(function(resolve) {
+      Testem.on('testem:next-module-response', function getTestModule(moduleName) {
+        if (moduleName !== undefined) {
+          loadIndividualModule(moduleName);
+        }
+
+        // if no tests were added, request the next module
+        if (qunit.config.queue.length === 0) {
+          Testem.emit('get-next-module');
+        } else {
+          Testem.removeEventCallbacks('testem:next-module-response', getTestModule);
+          resolve();
+        }
+      });
+      Testem.emit('get-next-module');
+    });
+  });
+  qunit.moduleDone(function() {
+    return new self.Promise(function(resolve) {
+      Testem.on('testem:next-module-response', function getTestModule(moduleName) {
+        if (moduleName !== undefined) {
+          loadIndividualModule(moduleName);
+        }
+
+        // if no tests were added, request the next module
+        if (qunit.config.queue.length === 0) {
+          Testem.emit('get-next-module');
+        } else {
+          // `removeCallback` removes if the event queue contains the same callback for an event.
+          Testem.removeEventCallbacks('testem:next-module-response', getTestModule);
+          Testem.removeEventCallbacks('testem:module-queue-complete', resolve);
+          resolve();
+        }
+      });
+      Testem.on('testem:module-queue-complete', resolve);
```

Comment:
I believe we leak event listeners here, specifically: if `testem:module-queue-complete` is triggered without `next-module-response` being triggered, who will remove the `testem:next-module-response` when we are done with it?